src/streamer/sObjectStream.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Org, StatusResult } from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { AnyJson, JsonMap, ensureJsonMap } from '@salesforce/ts-types';
+import DOCeStreaming from './doceStream';
+
+export default abstract class SObjectStreaming extends DOCeStreaming {
+  private idsToInspect: string[];
+  private channelName: string;
+  private matchProcessor: (message: JsonMap) => StatusResult;
+
+  protected constructor(org: Org, wait: Duration, idsToInspect: string[], channelName: string) {
+    super(org, wait);
+    this.idsToInspect = idsToInspect;
+    this.channelName = channelName;
+  }
+
+  /**
+   * This method sets the processor we will need inside the matchingProcessor and will start the stream.
+   *
+   * @param matchProcessor This processor will be used in the matchingProcessor, it's provided by each implementation.
+   */
+  protected async watchForSObject(matchProcessor: (message: JsonMap) => StatusResult): Promise<void | AnyJson> {
+    this.matchProcessor = matchProcessor;
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    return this.startStream(this.channelName, this.matchingProcessor.bind(this));
```

Comment:
This feels a bit wrong setting matchProcessor on the instance.  It assumes it will not change.  I think that is valid for today, but maybe we can protect against it
```suggestion
    return this.startStream(this.channelName, (event: JsonMap) => {
      return this.matchingProcessor(event, matchProcessor);
    });
```

Reply:
Great !!!
And this is even easier to test, because the method we pass on the parameter will not be bounded I think !

Reply:
Nope. is not easier to test the function we pass on parameter =/